.github/workflows/helm-chart.yml

```diff
@@ -16,14 +16,23 @@ jobs:
     steps:
       - uses: actions/checkout@v4
         with:
-          submodules: 'recursive'
+          submodules: "recursive"
       - name: "Extract NIX_PATH"
         id: nixpath
         run: |
           echo "nix_path=nixpkgs=$(jq -r '.nixpkgs.url' nix/sources.json)" >> $GITHUB_OUTPUT
       - uses: cachix/install-nix-action@v31
         with:
           nix_path: "${{ steps.nixpath.outputs.nix_path }}"
+          install_options: --no-daemon
+          extra_nix_config: |
+            keep-env-derivations = true
+            keep-outputs = true
+      - name: Restore and save Nix store
+        uses: nix-community/cache-nix-action@v6
+        with:
+          primary-key: helm-nix-${{ hashFiles('deploy/helm/shell.nix', 'nix/*') }}
```

Comment:
how is garbage collection done?

Reply:
Post Restore phase:

    The cache-nix-action purges caches whose keys are the same as the primary key and that were created or last accessed more than a given time ago (inputs: purge, purge-created, purge-last-accessed, purge-primary-key).

    When there's no cache whose key is the same as the primary key, the cache-nix-action collects garbage in the Nix store and saves a new cache (inputs: save, gc-max-store-size, gc-max-store-size-linux, gc-max-store-size-macos).

    The cache-nix-action purges caches whose keys match some of the given prefixes in a given list of key prefixes and that were created or last accessed more than a given time ago relative to the start of the Post Restore phase (purge, purge-prefixes, purge-created, purge-last-accessed, purge-primary-key).